visualization-design.typ

```diff
@@ -105,15 +105,126 @@ room = {
+enabled           = true
+
+# Disable total stats (stat_totals_*) metrics to be generated
+disable_total_stats = false
+
+# basic_auth_username =
+# basic_auth_password =
+```
+
+Setting a username and password is optional, only set if you want to require authorization to view the metrics endpoint. After the configuration file is edited as shown above and Grafana is restarted, the metrics should be accessable at http://localhost:3000/metrics.
+
+Next, add the job to your prometheus.yml file. Example:
+
+```yaml
+- job_name: 'grafana_metrics'
+
+   scrape_interval: 15s
+   scrape_timeout: 5s
+
+   static_configs:
+     - targets: ['localhost:3000']
+```
+
+This example job instructs Prometheus to scrape from port 3000 every 15 seconds. After adding the job, restart Prometheus and it should appear on the targets tab.
+
+Finally in Grafana, hover your mouse over the Configuration (gear) icon on the left sidebar and then click Data Sources. Select the Prometheus data source, and on the Dashboards tab, import the Grafana metrics dashboard. All scaped metrics should now be available.
+
+=== Setup Instructions From Prometheus
+
+To create a Prometheus data source in Grafana:
 
-- Querying Prometheus with Grafana: Grafana supports querying Prometheus, and there is documentation linked below on a quick start for creating a new data source. Given the balancer has already integrated Prometheus for metrics, this is the preferred method for gathering data. There is no mention on compatibility with D3.js, but assuming there are no issues integrating D3.js into a Grafana panel, this should not be a problem. #cite(<grafana-prometheus-visualization>)
+- Click on the "cogwheel" in the sidebar to open the Configuration menu.
+- Click on "Data Sources".
+- Click on "Add data source".
+- Select "Prometheus" as the type.
+- Set the appropriate Prometheus server URL (for example, http://localhost:9090/)
+- Adjust other data source settings as desired (for example, choosing the right Access method).
+- Click "Save & Test" to save the new data source.
+
+To create a new Grafana graph:
+
+- Click the graph title, then click "Edit".
+- Under the "Metrics" tab, select your Prometheus data source (bottom right).
+- Enter any Prometheus expression into the "Query" field, while using the "Metric" field to lookup metrics via autocompletion.
+- To format the legend names of time series, use the "Legend format" input. For example, to show only the method and status labels of a returned query result separated by a dash, you could use the legend format string {{method}} - {{status}}.
+- Tune other graph settings until you have a working graph.
+
+== Recieving Information From Load Balancers
+
+The data format specified in Data Format is in JSON format. Grafana is a tool primarily meant for time series data, and does not natively support JSON data sources. To add support, the JSON plugin must be installed:
+
+```cmd
+grafana-cli plugins install simpod-json-datasource
```

Comment:
I'm not confident this plugin is sufficient for what we want to do, at least on its own. It looks like it can only query 1 endpoint, which does not meet our requirements. We need to query state from all of the balancers, without knowing ahead of time what network address they have (similar to monolith discovery), then aggregate it all together into one representation.